### PR TITLE
docs(store): improve walkthrough instructions

### LINKED ISSUE #3128: Store Walkthrough missing types
This one comes from a [StackOverflow question](https://stackoverflow.com/questions/68914356/what-does-props-book-mean-in-ngrx-angular). It seems like our [walkthrough](https://ngrx.io/guide/store/walkthrough) is missing some type definitions for some [book actions](https://github.com/ngrx/platform/blob/master/projects/ngrx.io/content/examples/testing-store/src/app/state/books.actions.ts).

This should become

```ts
export const addBook = createAction(
  '[Book List] Add Book',
  props<{ bookId:number }>()
);

export const removeBook = createAction(
  '[Book Collection] Remove Book',
  props<{ bookId:number }>()
);

export const retrievedBookList = createAction(
  '[Book List/API] Retrieve Books Success',
  props<{ book:Book }>()
);
```

Because the book property name of the `retrievedBookList` action is now lowercase, all references to it should also be updated.

## I would be willing to submit a PR for the docs :heart:

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

### PR DESCRIPTION
**Pull request by mauromattos00**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

On the [@ngrx/store walkthrough](https://ngrx.io/guide/store/walkthrough), the book actions props do not have explicit types.

image

Closes #3128

## What is the new behavior?

Prop types explicitly declared to book actions.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by mauromattos00** [comment]
Hi, I just found out that I should not use "adds" not "removes" in the commit subject. I'm so sorry about this! Should I redo my PR with the right subject?

File: projects/ngrx.io/content/examples/store-walkthrough/src/app/models/book.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export interface Book {
```
Comment (timdeschryver) [inline projects/ngrx.io/content/examples/store-walkthrough/src/app/models/book.ts:1]: There's already a `Book` interface that we should reuse [here](https://github.com/ngrx/platform/blob/master/projects/ngrx.io/content/examples/store-walkthrough/src/app/book-list/books.model.ts). If we reuse that model, we should move it to this file, and remove the old file,

**Review by timdeschryver — CHANGES_REQUESTED** [review]
@mauromattos00 you can leave it as is, we can reword the commit when we merge it.

File: projects/ngrx.io/content/examples/store-walkthrough/src/app/models/book.ts (hunk 1 above)
Comment (mauromattos00) [reply projects/ngrx.io/content/examples/store-walkthrough/src/app/models/book.ts]: Oh, I didn't see it 😄
Done! Thanks for the tip

**Review by markostanimirovic — COMMENTED** [review]
@mauromattos00 

Can you revert changes from `book-list.component.ts` and `books.model.ts` files? 🙂

File: projects/ngrx.io/content/examples/store-walkthrough/src/app/state/books.actions.ts [hunk 2]
```diff
@@ -1,16 +1,17 @@
 import { createAction, props } from '@ngrx/store';
+import { Book } from '../book-list/books.model';
 
 export const addBook = createAction(
   '[Book List] Add Book',
-  props<{ bookId }>()
+  props<{ bookId: number }>()
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/examples/store-walkthrough/src/app/state/books.actions.ts:6]:
```suggestion
  props<{ bookId: string }>()
```

File: projects/ngrx.io/content/examples/store-walkthrough/src/app/state/books.actions.ts [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 );
 
 export const removeBook = createAction(
   '[Book Collection] Remove Book',
-  props<{ bookId }>()
+  props<{ bookId: number }>()
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/examples/store-walkthrough/src/app/state/books.actions.ts:11]:
```suggestion
  props<{ bookId: string }>()
```

**Review by markostanimirovic — CHANGES_REQUESTED** [review]

**Review by markostanimirovic — APPROVED** [review]

**Comment by mauromattos00** [comment]
> @mauromattos00
> 
> Can you revert changes from `book-list.component.ts` and `books.model.ts` files? 🙂

@markostanimirovic 

Done! 😀

**Review by timdeschryver — COMMENTED** [review]
Thanks @mauromattos00 !
After reading the updated docs, I think we should mention the `Book` interface in step 1/2. Because I'm afraid that it can cause some confusion, since we're using an interface that we don't mention anywhere.

Thoughts?
If we agree, we can append it to this PR, or create a second PR.

**Comment by mauromattos00** [comment]
@timdeschryver  That's true! it's not clear where this model came from.

As the step 9 instructions only meNtion the `book-list` folder and the `book.ts` creation, I can turn step 9 into step 2 and increase the subsequent step numbers. I believe this order would help the user to understand the model usage.

What do you think about it?

**Review by timdeschryver — COMMENTED** [review]
Oh... I missed step 9 😅
For me, this can become step 1, if we change the wording a bit to mention that we'll use that model throughout the whole walkthrough.

If I'm not mistaken, all steps can also have number 1, normally the correct order is added when the markdown is processed.

**Comment by mauromattos00** [comment]
Hey @timdeschryver!

Thanks for the suggestions! I sent the book model creation to step 1 and made a couple of text changes to make more sense with this change.

I changed the step numbers to number 1 so it can control the step numbers automaticaly, but it didn't work. 

Let me know if it fits the documentation needs! 😃

File: projects/ngrx.io/content/guide/store/walkthrough.md [hunk 4]
```diff
@@ -4,9 +4,12 @@ The following example more extensively utilizes the key concepts of store to man
 
 ## Tutorial
 
-1.  Generate a new project using StackBlitz <live-example name="ngrx-start" noDownload></live-example>.
+1.  Generate a new project using StackBlitz <live-example name="ngrx-start" noDownload></live-example> and create a folder named `book-list` inside `app` folder. We will be using it soon to hold our book list component. For now, let's start with adding a file named `books.model.ts` to reference different aspects of a book in the book list.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/walkthrough.md:7]: Suggestion: reword the sentence to not say "will"

```suggestion
1.  Generate a new project using StackBlitz <live-example name="ngrx-start" noDownload></live-example> and create a folder named `book-list` inside the `app` folder. This folder is used to hold the book list component later in the tutorial. For now, let's start with adding a file named `books.model.ts` to reference different aspects of a book in the book list.
```

File: projects/ngrx.io/content/guide/store/walkthrough.md [hunk 5]
```diff
@@ -4,9 +4,12 @@ The following example more extensively utilizes the key concepts of store to man
...
 
-2.  Right click on the `app` folder in StackBlitz. Create a state management folder `state`, and within the new folder, name a new file `books.actions.ts` to describe the book actions. Book actions include the book list retrieval, and the add and remove book actions.
+<code-example header="src/app/book-list/books.model.ts" path="store-walkthrough/src/app/book-list/books.model.ts">
+</code-example>
+
+2.  Right click on the `app` folder, create a state management folder `state`, and within the new folder, name a new file `books.actions.ts` to describe the book actions. Book actions include the book list retrieval, and the add and remove book actions.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/walkthrough.md:12]: Suggestion: break up the long sentence into two shorter ones

```suggestion
2.  Right click on the `app` folder to create a state management folder `state`. Within the new folder, create a new file `books.actions.ts` to describe the book actions. Book actions include the book list retrieval, and the add and remove book actions.
```

File: projects/ngrx.io/content/guide/store/walkthrough.md [hunk 6]
```diff
@@ -41,41 +44,37 @@ The following example more extensively utilizes the key concepts of store to man
...
-</code-example>
 
-10. In the same folder, we will have a service that fetches the data needed for the book list from an API. Create a file in the `book-list` folder named `books.service.ts`, which will call the Google Books API and return a list of books.
+9. In the `book-list` folder, we will have a service that fetches the data needed for the book list from an API. Create a file in the `book-list` folder named `books.service.ts`, which will call the Google Books API and return a list of books.
 
 <code-example header="src/app/book-list/books.service.ts" path="store-walkthrough/src/app/book-list/books.service.ts">
 </code-example>
 
-11. In the `book-list` folder, create the `BookListComponent` with the following template. Update the `BookListComponent` class to dispatch the `add` event.
+10. In the `book-list` folder, create the `BookListComponent` with the following template. Update the `BookListComponent` class to dispatch the `add` event.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/walkthrough.md:53]: Suggestion, to have a different start than the updated step 9

```suggestion
10. In the same folder (`book-list`), create the `BookListComponent` with the following template. Update the `BookListComponent` class to dispatch the `add` event.
```

File: projects/ngrx.io/content/guide/store/walkthrough.md [hunk 7]
```diff
@@ -41,41 +44,37 @@ The following example more extensively utilizes the key concepts of store to man
 <code-example header="src/app/state/books.selectors.ts" path="store-walkthrough/src/app/state/books.selectors.ts">
 </code-example>
 
-9.  Create a folder named `book-list`, which we will be using soon to hold our book list component. For now, let's start with adding a file named `books.model.ts` to reference different aspects of a book in the book list.
-
-<code-example header="src/app/book-list/books.model.ts" path="store-walkthrough/src/app/book-list/books.model.ts">
-</code-example>
 
-10. In the same folder, we will have a service that fetches the data needed for the book list from an API. Create a file in the `book-list` folder named `books.service.ts`, which will call the Google Books API and return a list of books.
+9. In the `book-list` folder, we will have a service that fetches the data needed for the book list from an API. Create a file in the `book-list` folder named `books.service.ts`, which will call the Google Books API and return a list of books.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/walkthrough.md:48]: Suggestion: reword the sentence to not say "will"

```suggestion
9. In the `book-list` folder, we want to have a service that fetches the data needed for the book list from an API. Create a file in the `book-list` folder named `books.service.ts`, which will call the Google Books API and return a list of books.
```

**Review by timdeschryver — COMMENTED** [review]
I find the updated version better, thanks @mauromattos00 .
I also left a couple of comments to reword some sentences, let me know what you think.

**Comment by mauromattos00** [comment]
Great suggestions @timdeschryver, the updates make the instructions more intuitive and fluid. Thanks!

Let me know if it needs any more changes. 🙂

**Review by timdeschryver — APPROVED** [review]
LGTM 👍

**Comment by brandonroberts** [comment]
Thanks @mauromattos00!

**Pull request merged**